### PR TITLE
add v21.1.0-rc.2 release note

### PR DESCRIPTION
**Pull request by taroface**
v21.1.0-rc.2 is scheduled for May 10.

See: https://github.com/cockroachdb/cockroach/issues/64578

### CODE REVIEW

File: releases/v21.1.0-rc.2.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+### Docker image
+
+{% include copy-clipboard.html %}
+~~~shell
+$ docker pull cockroachdb/cockroach-unstable:v21.1.0-rc.2
+~~~
+
+### Enterprise edition changes
+
+- [Changefeeds](../v21.1/change-data-capture.html) more reliably fail when [`IMPORT INTO`](../v21.1/import-into.html) is run against a targeted table. [#64372][#64372]
```
Comment (mikeCRL) [inline releases/v21.1.0-rc.2.md:35]:
```suggestion
- [Changefeeds](../v21.1/change-data-capture.html) now reliably fail when [`IMPORT INTO`](../v21.1/import-into.html) is run against a targeted table, as change data capture is [not supported](../v21.1/known-limitations.html#change-data-capture) for this action. [#64372][#64372]
```

I know the dev wrote "more reliably fail" but in that doc I've linked and [this comment](https://github.com/cockroachdb/cockroach/issues/43784#issuecomment-828816650) it seems that we expect this to always fail, as we've intended.

File: releases/v21.1.0-rc.2.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+### Bug fixes
+
+- Fixed a correctness bug, which caused partitioned index scans to omit rows where the value of the first index column was `NULL`. This bug was present since v19.2.0. [#64046][#64046]
+- [`IMPORT`](../v21.1/import.html) and [`RESTORE`](../v21.1/restore.html) jobs that were in progress during a cluster backup will now be canceled when that cluster backup is restored. This fixes a bug where these restored jobs may have assumed to make progress that was not captured in the backup. [#64352][#64352]
```
Comment (mikeCRL) [inline releases/v21.1.0-rc.2.md:40]:
```suggestion
- [`IMPORT`](../v21.1/import.html) and [`RESTORE`](../v21.1/restore.html) jobs that were in progress during a cluster backup will now be canceled when that cluster backup is restored. This fixes a bug where these restored jobs may have recorded prior progress that was not captured in the backup. [#64352][#64352]
```

Big caveat: Not sure if this is accurate as the original PR is a bit hard to read, however, I think this needed a touch more clarity (e.g. who is assuming and when) so please consider this edit *if* you think it's accurate.

File: releases/v21.1.0-rc.2.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+- Fixed a race condition where read-only requests during replica removal (e.g., during range merges or rebalancing) could be evaluated on the removed replica, returning an empty result. [#64370][#64370]
```
Comment (mikeCRL) [inline releases/v21.1.0-rc.2.md:41]:
```suggestion
- Fixed a race condition where read-only requests during replica removal (for example, during range merges or rebalancing) could be evaluated on the removed replica, returning an empty result. [#64370][#64370]
```

**Review by mikeCRL — APPROVED** [review]
Just some minor suggestions

**Review by taroface — COMMENTED** [review]
TFTR! Added one new release note for your review.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/10507#-:-M_MxxVV6M_DowRa33yT:b-q5cvvd)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @mikeCRL)

___
*[releases/v21.1.0-rc.2.md, line 35 at r1](https://reviewable.io/reviews/cockroachdb/docs/10507#-MZwxxvKAeXvICHiBvaT-r1-35:-M_MwvMAEYYzKzQmKzd7:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/f118a468f61ad689da9b3231d9a6a343617477c1/releases/v21.1.0-rc.2.md#L35)):*
<details><summary><i>Previously, mikeCRL (Mike Lewis) wrote…</i></summary><blockquote>

```suggestion
- [Changefeeds](../v21.1/change-data-capture.html) now reliably fail when [`IMPORT INTO`](../v21.1/import-into.html) is run against a targeted table, as change data capture is [not supported](../v21.1/known-limitations.html#change-data-capture) for this action. [#64372][#64372]
```

I know the dev wrote "more reliably fail" but in that doc I've linked and [this comment](https://github.com/cockroachdb/cockroach/issues/43784#issuecomment-828816650) it seems that we expect this to always fail, as we've intended.
</blockquote></details>

Done.
___
*[releases/v21.1.0-rc.2.md, line 40 at r1](https://reviewable.io/reviews/cockroachdb/docs/10507#-MZwxxvKAeXvICHiBvaT-r1-40:-M_MxPcX1OrOg_Lmwdkq:bcx80j4) ([raw file](https://github.com/cockroachdb/docs/blob/f118a468f61ad689da9b3231d9a6a343617477c1/releases/v21.1.0-rc.2.md#L40)):*
<details><summary><i>Previously, mikeCRL (Mike Lewis) wrote…</i></summary><blockquote>

```suggestion
- [`IMPORT`](../v21.1/import.html) and [`RESTORE`](../v21.1/restore.html) jobs that were in progress during a cluster backup will now be canceled when that cluster backup is restored. This fixes a bug where these restored jobs may have recorded prior progress that was not captured in the backup. [#64352][#64352]
```

Big caveat: Not sure if this is accurate as the original PR is a bit hard to read, however, I think this needed a touch more clarity (e.g. who is assuming and when) so please consider this edit *if* you think it's accurate.
</blockquote></details>

Hard to tell from the PR - going to keep the original wording as I think it is safer, if more awkward.
___
*[releases/v21.1.0-rc.2.md, line 41 at r1](https://reviewable.io/reviews/cockroachdb/docs/10507#-MZwxxvKAeXvICHiBvaT-r1-41:-M_MxroT3HOYXXPe3dbU:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/f118a468f61ad689da9b3231d9a6a343617477c1/releases/v21.1.0-rc.2.md#L41)):*
<details><summary><i>Previously, mikeCRL (Mike Lewis) wrote…</i></summary><blockquote>

```suggestion
- Fixed a race condition where read-only requests during replica removal (for example, during range merges or rebalancing) could be evaluated on the removed replica, returning an empty result. [#64370][#64370]
```
</blockquote></details>

Done.

**Comment by mikeCRL** [comment]
@taroface Additional commit LGTM

**Pull request merged**